pykechain/models/property_reference.py

```diff
@@ -357,14 +358,19 @@ def upload(self, data: Any) -> None:
         :raises OSError: When the path to the file is incorrect or file could not be found
         """
         filename = os.path.basename(data)
-        stored_file = self._client.create_stored_file(
+        new_stored_file = self._client.create_stored_file(
             name=filename,
             scope=self.scope,
             classification=StoredFileClassification.SCOPED,
             category=StoredFileCategory.REFERENCED,
             filepath=data,
         )
-        self.value = self.value + [stored_file] if self.value else [stored_file]
+        if self.has_validator(SingleReferenceValidator):
+            self.value = [new_stored_file]
+        else:
+            self.value = (
+                self.value + [new_stored_file] if self.value else [new_stored_file]
+            )
```

Comment:
2 small issues with this:
- this code has subtle purpose (replacing `self.value`) it's difficult to get at first glance. Can we add a small comment explicitly describing value replacement is important?  I had to stop for a moment because i confused it with list comprehension and rewrote this comment 3 times before i fully got the purpose of code
- `self.value` seems to always be a list

Therefore I think this whole `if/else` block should work equivalently:
```suggestion
        if self.has_validator(SingleReferenceValidator):
            # we want to replace value
            self.value = [new_stored_file]
        else:
            self.value.append(new_stored_file)
```

If not then may I ask to unwind this oneliner to:
```suggestion
        if self.has_validator(SingleReferenceValidator):
            # we want to replace value
            self.value = [new_stored_file]
        else:
            if self.value:
                self.value.append(new_stored_file)
            else:
                self.value = [new_stored_file]
```